packages/opencensus-exporter-prometheus/src/prometheus-stats.ts

```diff
@@ -0,0 +1,217 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AggregationType, DistributionData, ExporterConfig, logger, Logger, Measurement, MeasureUnit, StatsEventListener, Tags, View} from '@opencensus/core';
+import * as express from 'express';
+import * as http from 'http';
+import {Counter, Histogram, Gauge, register, Metric} from 'prom-client';
+
+export interface PrometheusExporterOptions extends ExporterConfig {
+  /** App prefix for metrics, if needed - default opencensus */
+  prefix?: string;
+  /**
+   * Port number to Prometheus server
+   * Default registered port is 9464:
+   * https://github.com/prometheus/prometheus/wiki/Default-port-allocations
+   */
+  port?: number;
+  /** Define if the Prometheus server will be started - default false */
+  startServer?: boolean;
+}
+
+/** Format and sends Stats to Prometheus */
+export class PrometheusStatsExporter implements StatsEventListener {
+  static readonly defaultOptions = {
+    port: 9464,
+    startServer: false,
+    contentType: 'text/plain; text/plain; version=0.0.4; charset=utf-8',
+    prefix: 'opencensus'
+  };
+
+  private logger: Logger;
+  private prefix: string;
+  private port: number;
+  private app = express();
+  private server: http.Server;
+
+  constructor(options: PrometheusExporterOptions) {
+    this.logger = options.logger || logger.logger();
+    this.port = options.port || PrometheusStatsExporter.defaultOptions.port;
+    this.prefix =
+        options.prefix || PrometheusStatsExporter.defaultOptions.prefix;
+
+    /** Start the server if the startServer option is true */
+    if (options.startServer) {
+      this.startServer();
+    }
+  }
+
+  /**
+   * Not used in this context
+   * @param view
+   */
+  onRegisterView(view: View) {}
+
+  /**
+   * Method called every new stats' record
+   * @param views
+   * @param measurement
+   */
+  onRecord(views: View[], measurement: Measurement) {
+    for (const view of views) {
+      this.updateMetric(view, measurement);
+    }
+  }
+
+  /**
+   * Register or get a metric in Prometheus
+   * @param view View will be used to register the metric
+   * @param tags Optional, used only for histogram metric
+   */
+  private registerMetric(view: View, tags?: Tags): Metric {
+    const metricName = this.getPrometheusMetricName(view);
+    /** Get metric if already registered */
+    const metric = register.getSingleMetric(metricName);
+
+    // Create a new metric if there is no one
+    if (!metric) {
+      const metricObj = {
+        name: metricName,
+        help: view.description,
+        labelNames: view.getColumns(),
+      };
+      // Creating the metric based on aggregation type
+      switch (view.aggregation) {
+        case AggregationType.COUNT:
+          return new Counter(metricObj);
```

Comment:
Yes, because this new instance run the registerMetric in background. Anyway, I'm going to change this to fit with the proposed changes in your first comment.